src/components/modals/share-modal/index.tsx

```diff
@@ -3,15 +3,25 @@ import Modal from 'react-bootstrap/Modal';
 import { useRouter } from 'next/router';
 import useAppSelector from '../../../store/hooks/useAppSelector';
 import { selectAllItems, selectById } from '../../../features/leda-nft/store/leda-nft.slice';
+import { Item } from '../../../types/item';
 
 type Props = {
   show: boolean;
   handleModal: () => void;
+  itemId?: string;
 };
 
-const ShareModal = ({ show, handleModal }: Props) => {
+const ShareModal = ({ show, handleModal, itemId }: Props) => {
   const { query } = useRouter();
-  const item = useAppSelector((state) => selectById(state, `${query.itemId}`));
+  const itemQuery = useAppSelector((state) => selectById(state, `${query.itemId}`));
+  const itemByProps = useAppSelector((state) => selectById(state, `${itemId}`));
```

Comment:
Minor. You could make only once call, and pass whatever value is defined. According to your logic, the itemId by props has higher priority (the last if condition). So I would do it like `useAppSelector((state) => selectById(state, itemId || query.itemId)); `. Make sure to validate undefined values in your selector so you don't break it.

Reply:
I didn't think that way. I'll keep it in mind